ingestion/src/levels.rs

```diff
@@ -145,24 +211,40 @@ pub fn param_get_level(
 ) -> Result<Option<i32>, Error> {
     let level_table = level_table.read().map_err(|e| Error::Lock(e.to_string()))?;
 
-    if let Some(param_level) = level_table.get(&param_id) {
-        // if level passed into this function is 0, replace with default from stinfosys
-        let lvl = match level {
-            Some(0) => param_level.hlevel, // this is the default
-            Some(lvl) => lvl,              // keep the value
-            None => return Ok(None),
-        };
-        // Convert level to cm (use scale from stinfosys)
-        // scale for things that are currently in m is 0, so need to shift
-        // could be that we do not have a scale, or encounter ones we have not currently accounted for
-        let lvl = match param_level.hlevel_scale {
-            Some(0) => lvl * 100, // convert from meters
-            Some(-2) => lvl,      // already in cm
-            // oh dear, this isn't meters or cm?
-            _ => unreachable!(), // this shouldn't happen due to check at import!
-        };
-        return Ok(Some(lvl));
+    // Since we have filled in things from stinfosys as long as we found a scale
+    // this means no scale existed for this param, and thus it cannot be used
+    // since we cannot convert it, and be sure it has the right units.
+    // Thus our only option (if we want to keep this data) is to insert it with
+    // NULL, and eventually have it corrected by content managers.
+    let Some(param_level) = level_table.get(&param_id) else {
+        warn!("could not find a scale for this param: {param_id}");
+        return Ok(None);
+    };
+
+    // If input level is already NULL, we simply insert NULL
+    // however, this should never happen since kvalobs / obsinn have default 0
+    let Some(mut lvl) = level else {
+        return Ok(None);
+    };
```

Comment:
This is contingent on my change to the obsinn/checked parsers in the kafka raw PR that substitutes 0 where sensor level is missing. If you're going to remigrate based on this you should probably cherry-pick that commit

Reply:
https://github.com/metno/lard/pull/113/commits/6029b9335be65c0e3fe33d41d07820f8f0c3e5c9

Reply:
I guess then we need `let mut lvl = level.unwrap_or(0)` here, or we could have it outside while making the function accept directly an `i32`